### PR TITLE
System Redesign

### PR DESCRIPTION
**Pull request by tanvirtin**
Due to how coupled things are in the existing system and the incorrect design decisions taken by me, the project was becoming difficult to maintain and enhance for the future, moreover it was getting much harder to implement features that Neovim users truly deserve.

It's a lot of work but, it's something this project needs to push it forward as a good Git integration plugin for Neovim.

**This PR will break several existing Neovim config and I apologize for it** 🙏🏾

If you want to try these new features switch to the **develop** branch.

This pull request will break a bunch of things, but it's growing pains, which introduces a bunch of new enhancements and existing bug fixes.

New Features:
- Much better user interface
- Much more performant
- Users can now cycle through the table view
- Reduced bloat from project
- Split diff view in hunk preview
- Ability for users to perform existing commands across Project Diff Preview.
- VGit now recognizes git files outside of the project's working directory
- VGit will now recognize if a file has been renamed.
- New preview called `project_hunks_preview` to open a quick preview of all the hunks in your entire project which looks similar to the quickfix preview, but with the VGit `CodeComponent` baked inside it. Allowing you to quickly see all the diffs in your project in both split and unified diff views.
- Ton of quality life improvements

Removed Features:
- diff base for gutter (current implementation makes the system very convoluted to use, e.g VSCode doesn't even support this)
- telescope integration (unnecessary bloat)
- Removed theming support (useless features makes makes codebase messy and complex)
- Removed layout support (too much configuration makes code harder to maintain)
- Renamed `buffer_staged_diff_preview` to `buffer_diff_staged_preview`

Closes issues:
- #120 
- #119 
- #100 
- #79
- #145 
- #155
- #154
- #135

### CODE REVIEW

**Comment by Shatur** [comment]
Since the PR is already a breaking change, can I suggest to use a more common highlighting groups for signs?
Currently we have `VGitSignAdd`, `VGitSignRemove` and `VGitSignChange` which are vgit-specific. But it would be nice to use more common names such as `GitSignsAdd`, `GitSingsDelete` and `GitSignsChange` as in `gitsigns` plugin. Such things makes live easier for colorscheme developers.

File: lua/vgit/settings/hls.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  GitSignAdd = {
+    fg = '#d7ffaf',
+  },
+  GitSignChange = {
+    fg = '#7AA6DA',
+  },
+  GitSignDelete = {
+    fg = '#e95678',
+  },
```
Comment (Shatur) [inline lua/vgit/settings/hls.lua:25]: Maybe use `GitSigns*` (plural)?
This is how it named in [gitsigns](https://github.com/lewis6991/gitsigns.nvim/blob/main/lua/gitsigns/highlight.lua#L27).
```suggestion
  GitSignsAdd = {
    fg = '#d7ffaf',
  },
  GitSignsChange = {
    fg = '#7AA6DA',
  },
  GitSignsDelete = {
    fg = '#e95678',
  },
```

**Comment by Shatur** [comment]
Can we also close #157 with this?

**Comment by tanvirtin** [comment]
With these changes paving the way for version `v0.1.0`, I am super hopeful to get some potential future contributors who can enhance this plugin further. Going to leave #157 for anyone to pickup and contribute, it should be fairly simple!

**Pull request merged**